src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -709,24 +731,22 @@ private class NativeMethods
             internal static extern ErrorCode rd_kafka_position(
                     IntPtr rk, IntPtr partitions);
 
+            // note: producev signature is rd_kafka_producev(rk, ...)
+            // __arglist is an undocumented keyword which deal with it
+            // but doesn't work on standardlib (and undocumented so don't use)
+            // stay simple with one binding for now, but it will be worth benchmarking
+            // the overload with no timestamp, opaque, partition, etc
             [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-            internal static extern IntPtr rd_kafka_produce(
-                    IntPtr rkt,
```

Comment:
it seems a pity to require a memory allocation + string serialization for every produce call now with producev :-( - @edenhill would be possible to make the `rd_kafka_producev` method alternatively take a topic handle? i guess this is super low priority - the overall perf hit will be negligible (though i think it's probably measurable).